packages/@contentful--app-scripts/lib/upload/create-zip-from-directory.js

```diff
@@ -0,0 +1,61 @@
+const glob = require('glob');
+const chalk = require('chalk');
+const fs = require('fs');
+const JSZip = require('jszip');
+const { showCreationError } = require('../utils');
+
+async function convertPathToFileData(filePath) {
+  return new Promise((res) => {
+    fs.readFile(filePath, 'utf8', (err, data) => {
+      res([filePath, data]);
+    });
+  });
+}
+
+function addFilesToZip(files, zip, path) {
+  files.forEach(([filePath, data]) => {
+    // remove the root path to not have the folder inside the zip
+    let localPath = filePath;
+    const firstIndex = filePath.indexOf(path);
+    if (firstIndex === 0) {
+      localPath = localPath.substring(localPath.length, path.length);
+    }
+    zip.file(localPath, data);
+  });
+  return zip;
+}
+
+async function createZipFileFromDirectory(path) {
+  try {
+    let zip = new JSZip();
+    const filePaths = await new Promise((res) => {
```

Comment:
You can use `util.promisify` here to stick with the async/await style we have elsewhere